mapper/javabean/src/main/java/org/hibernate/search/mapper/javabean/session/SearchSession.java

```diff
@@ -9,27 +9,66 @@
 import java.util.Collection;
 import java.util.Collections;
 
+import org.hibernate.search.engine.search.dsl.query.SearchQueryContext;
 import org.hibernate.search.mapper.javabean.search.SearchScope;
 import org.hibernate.search.mapper.javabean.search.dsl.query.SearchQueryResultDefinitionContext;
 import org.hibernate.search.mapper.javabean.work.SearchWorkPlan;
 
 public interface SearchSession extends AutoCloseable {
 
+	/**
+	 * Execute any pending work in the {@link #getMainWorkPlan() main work plan}
+	 * and release any resource held by this session.
+	 */
 	@Override
 	void close();
 
+	/**
+	 * Initiate the building of a search query.
+	 * <p>
+	 * The query will target the indexes mapped to the given type, or to any of its sub-types.
+	 *
+	 * @param type An indexed type, or a supertype of all indexed types that will be targeted by the search query.
+	 * @return A context allowing to define the search query,
+	 * and ultimately {@link SearchQueryContext#toQuery() get the resulting query}.
+	 * @see SearchQueryResultDefinitionContext
+	 */
 	default <T> SearchQueryResultDefinitionContext search(Class<T> type) {
```

Comment:
Yes: generics in varargs lead to call-site warnings ("heap pollution yada yada"). Usually the solution to avoid these warnings is to add `@SafeVarargs`, but this can only be used on static or final methods, so not here in an interface.

For the JavaBean mapper, we could get rid of the generics, but not for the ORM mapper, where they are actually useful.

Our only other option would be to declare one method per number of parameters: one type, two types, etc. I'm really not thrilled by the idea.

Reply:
Hum. This is annoying :/.

What makes them useful for the ORM case?

Reply:
The generics, you mean? In short it allows for typed queries when requesting an entity result type. Without it, we wouldn't know which type to put in `SearchQuery<T>`.
It's not useful in the JavaBean mapper because it cannot load entities from the database, since there isn't any database.